File: TIA/tia_final_project/src/com/crdzbird/models/Path.java
package com.crdzbird.models;

import java.util.Objects;

public class Path implements Comparable<Path> {
    
    private final Integer from;
    private final Integer to;

    public Path(Integer from, Integer to) {
        this.from = from;
        this.to = to;
    }

    public Integer getFrom() {
        return from;
    }

    public Integer getTo() {
        return to;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (!(obj instanceof Path)) {
            return false;
        }
        Path other = (Path) obj;
        return Objects.equals(from, other.from) && Objects.equals(to, other.to);
    }

    @Override
    public int hashCode() {
        return Objects.hash(from, to);
    }

    @Override
    public int compareTo(Path o) {
        int cmp = from.compareTo(o.from);
        return cmp != 0 ? cmp : to.compareTo(o.to);
    }
}

File: TIA/tia_genetic_memetic/lib/models/path.dart
class Path implements Comparable<Path> {
  final int from;
  final int to;

  Path(this.from, this.to);

  int get getFrom => from;

  int get getTo => to;

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;

    if (other is Path) {
      return from == other.from && to == other.to;
    }

    return false;
  }

  @override
  int get hashCode => from.hashCode ^ to.hashCode;

  @override
  int compareTo(Path other) {
    int cmp = from.compareTo(other.from);
    return cmp != 0 ? cmp : to.compareTo(other.to);
  }
}
